datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -802,7 +802,7 @@ impl<'a, S: SimplifyInfo> TreeNodeRewriter for Simplifier<'a, S> {
                 && !info.get_data_type(&left)?.is_floating()
                 && is_zero(&right) =>
             {
-                return Err(DataFusionError::ArrowError(ArrowError::DivideByZero));
+                return plan_err!("Divide by zero");
```

Comment:
Is the idea that since the error wasn't actually generated by Arrow then we shouldn't return an Arrow error?

In theory having a structured `ArrowError` would be more specific than a general PlanError with a "Divide by zero" message, but I am not sure how much it matters 🤷

Reply:
Thanks @alamb for this specific I have doubled checked before modification the but there is no arrow kernel direct call, that was the reason to change